### DISCUSSION TITLE
MHI2_ER_AUG22_S2133  , SW0358  /  FEC 023000EE ?

### DISCUSSION THREAD

**Discussion by 70lancz** [Mr-MIBonk/M.I.B._More-Incredible-Bash#106] [Q&A]
Hi,
I have software version in Audi A4 2016  , MHI2_ER_AUG22_S2133    . I can  not find in  patch file the version S2133. M.I.B._More-Incredible-Bash-3.4.1 patch is installd ok but I can not find in GEM FEC 023000EE only FEC  from factory. New   creating new patch ?
Thans for answer.

**Comment by harman-f** [comment]
This is a quite old FW version, which is not supported by M.I.B.

You have to update FW 1st.
Get FW version MHI2_ER_AUG22_K2161and replace the metainfo2.txt - in root of SD card - with this one:
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10682872/metainfo2.txt)

Could you share your M.I.B backup?

**Reply by hiepvu-ds** [reply]
Hello Thank you so much for support us.

I have a audi Q7 2016 FW Mhi2_er_aug22_s2140. I would like to upgrade to k2161 to have aa/cp. But I saw some people get sound problem after upgrade. The solution is unselect bose amplifier before upgrade new fw. However i can not unselect ( only standard menu is active, use-defined is disable).

Could you help me to verify your metainfo2.txt can help to solve this problem. 

Thank you.

**Comment by 70lancz** [comment]
Hi.
I have replaced  metainfo2.txt. During update i become fauld :  RCC_Pre_Hashes.txt   cod: 119
I had 3 options .  repeat,skip device,cancel update. After repeat is to same . I canceled the update.
Then device reboot. Now i have in unit  MHI2_ER_AUG22_K2161     sw 1421.  But in MMI  RCC,IOC,MMX2,MuTnrRef /nok/
With GEM  all FEC installed but the added FEC are Illegal.
I don't know what can i do next.
How can i send backup files to You ? I tis to large 209 MB


[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10684415/MU1421-JTB-07412.01.1611760CC7.log)

**Comment by harman-f** [comment]
Cancel is always the wrong choice!
Luckily your unit still does boot.

You have to run the FW update again.
Now you can use the stock metainfo2.txt, as the train is already changed.

The error points toward a defective SD card or corrupt file on your SD card.
Get a fresh SD card and copy MHI2_ER_AUG22_K2161 on it again.
Start the FW update

**Reply by pawtys22** [reply]
Hello, I've done an update on my FW from MHI2_ER_AU57x_P3634 to 3663_1 (aim was to get CarPlay, 'cos in MMI Menu there is ASI but in grey meens inactive.. The car is: Audi A6 C7, allroad, Europe Germany  model 2017) but after approx 2 Mins I've got an error: rcc_pre hashes.txt code: 119. I cancelled the update, MMI has restarted The sort of summary page came out: there were some notOK, MMI restarted again automatickly I belive..Than I looked into MMI Info : I can see the actuall version shows 3663_1 up . As I checked later on
on my SD card which I used to mentioned Update the content of FW was not fully there! There was about 1,67 GB on it and it should be approx 8.5 GB I believe (Folders: 1&2 with content and three txt files). For now MMI opens up and looks like it works but ASI is still in grey. I believe the update of FW went wrong. Shall I repeat the update using the proper SD card? For any ideas I will be gratefull. Best Regards

**Comment by 70lancz** [comment]
Here is my metainfo2.txt from original map update from Audi .

[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10684533/metainfo2.txt)

**Comment by harman-f** [comment]
No idea what this helps in this context...

**Comment by 70lancz** [comment]
I have created a new SD card with original metainfo2.txt . Update done ok . I have instaled the FEC with GEM /add_feccontainer.fec/  .After reboot all FEC are legal.
But if I insert SD card with Maps/ new created/ , update can not find valid maps. Then if i check Red menu added FEC files become illegal.
 Next. If I go to  AIO patch then become ,, search File MU1421-ifs-root-part2-0x00c20000-A078E900.ifs     no flashfiles there! ,,

Maybe something miss ?
[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10686130/MU1421-JTB-07412.01.1611760CC7.log)

**Reply by harman-f** [reply]
Why not just run the AIO patch?

you also need to patch this FW, before adding new FECs. 
you have to clean the illegal FECs now. 
And run AIO patch.

**Reply by harman-f** [reply]
Maps will only work after patching.

**Comment by 70lancz** [comment]
i should clean illegal FECs with with GM patch ?

**Reply by harman-f** [reply]
No, there is a function in M.I.B to clean illegals FECs. 
you can patch now, but the illegals FECs will be visible in unit until you clean them out.

**Comment by 70lancz** [comment]
Now i have only legal FECs in MMI. I go create a new SD with patch .
  apply SVM fix reboot , then in  GEM  backup and AIO patch .
With AIO patch all new FECs i become in my MMI ? Or i must add manually ?

**Comment by 70lancz** [comment]
Here is my last log.
How do I know that MMI is fully patched?

Sorry that you have so much to do with me.

[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10688274/MU1421-JTB-07412.01.1611760CC7.log)

**Comment by harman-f** [comment]
No idea what you are doing here!

LOG shows that you seem to have still the ifs-root from MHI2_ER_AUG22_S2133 on unit.
1. Please install MHI2_ER_AUG22_K2161 again with this metainfo2.txt
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10690431/metainfo2.txt)

2. Creat a fresh M.I.B SD
3. Install M.I.B
4. share the backup
5. Run only AIO Patcher!
![image](https://user-images.githubusercontent.com/98130152/217667619-460f03de-0de1-485e-b521-0a08d90a07d6.png)

6. Share LOG file

Please only this steps ;)

**Comment by 70lancz** [comment]
Hi. I have created new SD and replaced metainfo2.txt
During Flash become 3 options /repeat,skip,cancel update/ repeat keeps repeating,skip skip to cancel
After cancel update fauld. /next picture/
In MMI i have and  had soft K2161
Log   not created in SD
![001](https://user-images.githubusercontent.com/124744323/217830967-0f5caeca-14f4-45fa-afa7-a473294334a8.jpg)
![002](https://user-images.githubusercontent.com/124744323/217831000-f62a7c67-6887-4b72-9b5d-be63e7320a22.jpg)
![003](https://user-images.githubusercontent.com/124744323/217831022-a7adb7fd-e27c-4f64-99be-435867f1b4a3.jpg)
![004](https://user-images.githubusercontent.com/124744323/217831038-8d22c606-3050-4187-82bc-0a0ddcc45842.jpg)
![005](https://user-images.githubusercontent.com/124744323/217831048-04700e77-ed3a-40d0-bd78-768918924e88.jpg)
Sorry.

**Reply by harman-f** [reply]
Which FW file did you download?

**Reply by harman-f** [reply]
Please share the original metainfo2 from that download.

**Reply by 70lancz** [reply]
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10698634/metainfo2.txt)

**Reply by 70lancz** [reply]
This is the replaced metainfo.

**Reply by 70lancz** [reply]
And next is from .......
 (https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10698647/metainfo2.txt)
 M.I.B._More-Incredible-Bash-3.4.1.zip\M.I.B._More-Incredible-Bash-3.4.1 - ZIP  
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10698651/metainfo2.txt)

**Reply by 70lancz** [reply]
Yesterday after replaced metainfo2 i put original metainfo2 from M.I.B._More-Incredible-Bash-3.4.1.zip  and update to K 2161 was dan ok. Without fould.

**Reply by 70lancz** [reply]
now i created SD with original metainfo2.txt from MIB.zip.
[Result](result: no fauld)
 
[Uploading MU1421-JTB-07412.01.1611760CC7.log…]()

**Reply by 70lancz** [reply]
Next log is after  run  GEM , AIO patch.
[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10699210/MU1421-JTB-07412.01.1611760CC7.log)

**Comment by 70lancz** [comment]
I have all FW in  SD.

**Comment by 70lancz** [comment]
This is how it happened the first flash /yesterday/ from S 2133 , after was replaced metainfo2.

**Comment by harman-f** [comment]
As mentioned above, what ever you have done, there is still the old ifs-root-stage2 on your unit, which is in addition not patched.
Clearly showing that at least this part and moste likely also other parts of the FW update did NOT work!

From your latest LOG:
`search File MU1421-ifs-root-part2-0x00c20000-A078E900.ifs`
A078E900 is the header checksum of the old FW and NOT MU1421-MHI2_ER_AUG22_K2161.

![image](https://user-images.githubusercontent.com/98130152/217953709-4a237a19-db04-4b48-b9e3-9b31f3689df3.png)
That the unit does show this, does NOT mean that the FW is fully installed.
This is changed at the very beginning of the FW update, even  if you cancel the update - like you did.

![image](https://user-images.githubusercontent.com/98130152/217952722-6667208b-f57c-4ebb-886a-56b761fdcb14.png)
States, that checksums within the FW update do not match.
This means, that:
1. Your SD card is defect
2. You have corrupted files on your SD card
3. You have not copied MU1421-MHI2_ER_AUG22_K2161 onto the SD card.

Before anything you have to make sure, that you fully install MU1421-MHI2_ER_AUG22_K2161 on your unit.
Make sure you also use a fresh SD card.

**Reply by 70lancz** [reply]
I have downloaded fresh MIB Bash 3.4.1 and fresh MIB Patchis.
- I formatted the new SD, also copied "MHI2_ER_AUG22_K2161_MU1421_PATCH",,
- i used metainfo2.txt from ,,M.I.B._More-Incredible-Bash-3.4.1,,
- the result
[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10706972/MU1421-JTB-07412.01.1611760CC7.log)
-if i use metainfo2.txt
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10707446/metainfo2.txt)
 I have no other option but to repeat, repeat.......or cancel the update ,,flash does not continue,,

I'm sorry

**Reply by Csall67** [reply]
You can try the REM menu,Update Firmware MU1421-MHI2_ER_AUG22_K2161.after.run MIB Bash 3.4.1 and fresh MIB Patchis.

**Reply by 70lancz** [reply]
Yes, I understood that. I just don't know how to finish the update MU1421-MHI2_ER_AUG22_K2161witout fauld.

**Reply by 70lancz** [reply]
ok i'll try it

**Reply by 70lancz** [reply]
Just i think , but i am not shure if I insert the SD cart with metainfo2 from MIB Bash 3.4.1 /not modificated/ i see in display after update begining only  Launcer is (Y) and oll other (N/A). Thet mean to update go only Launcer ?

**Reply by 70lancz** [reply]
With this metainfo2.txt are RCC,IOC,MuTnrRef..... all (Y) but soething makes it impossible to update .
[metainfo2.txt](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10709068/metainfo2.txt)

**Reply by Csall67** [reply]
Download the complete firmware file,Copy to the root directory.Update in REM menu.SD card does not contain MIB file.
(not MHI2_ER_AUG22_K2161_MU1421_PATCH,It's just a patch）

**Reply by 70lancz** [reply]
where can i find complete files ,,MHI2_ER_AUG22_K2161_MU1421,,.?
I don't want to do something wrong with bad software.
Thanks

**Reply by harman-f** [reply]
yes, please download this ;)

Using my custom metainfo2.txt will work with that download.

**Reply by 70lancz** [reply]
Ok .
I've been downloading it since yesterday, the download is very slow. 
Will coding be required after the software change?

**Reply by 70lancz** [reply]
Hi.
All done. The new maps are in the MMI.
I want to thank you very much for your support and help.
I really appreciate it.

[MU1421-JTB-07412.01.1611760CC7.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10716576/MU1421-JTB-07412.01.1611760CC7.log)
